gateways/core/resource/start.go

```diff
@@ -45,15 +49,22 @@ func (ce *ResourceConfigExecutor) StartConfig(config *gateways.ConfigContext) {
 
 	for {
 		select {
-		case <-config.StartChan:
-			ce.GatewayConfig.Log.Info().Str("config-name", config.Data.Src).Msg("configuration is running.")
-			config.Active = true
-
-		case data := <-config.DataChan:
-			ce.GatewayConfig.DispatchEvent(&gateways.GatewayEvent{
-				Src:     config.Data.Src,
-				Payload: data,
-			})
+		case _, ok := <-config.StartChan:
```

Comment:
this code blocks seems to be repeated a lot in all the gateway implementations.. can this be abstracted to a common func?

Reply:
yeah, I'll create a separate PR for this.